krate/src/main/kotlin/hu/autsoft/krate/validated/ValidatedPreferenceDelegate.kt

```diff
@@ -0,0 +1,28 @@
+package hu.autsoft.krate.validated
+
+import hu.autsoft.krate.Krate
+import hu.autsoft.krate.validated.exception.KrateValidationException
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * [ValidatedPreferenceDelegate] is a generic [ReadWriteProperty] that can be used to
+ * validate values that are being set.
+ *
+ * @param [delegate] the [ReadWriteProperty] implementation that is used for delegation
+ * @param [isValid] the lambda used to validate property values on [setValue]
+ */
+internal class ValidatedPreferenceDelegate<T>(
+        private val delegate: ReadWriteProperty<Krate, T>,
+        private val isValid: (T) -> Boolean
+) : ReadWriteProperty<Krate, T> by delegate {
```

Comment:
This delegation of the delegate to the delegate is amazing, well done! :)

Reply:
yo dawg i heard you like delegates, so I got delegates for your delegates on delegates.